File: 0617/0617.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */


struct TreeNode* mergeTrees(struct TreeNode* root1, struct TreeNode* root2) {
    if (root1 == NULL) {
        return root2;
    }
    if (root2 == NULL) {
        return root1;
    }
    struct TreeNode* mergedNode = malloc(sizeof(struct TreeNode));
    mergedNode->val = root1->val + root2->val;
    mergedNode->left = mergeTrees(root1->left, root2->left);
    mergedNode->right = mergeTrees(root1->right, root2->right);

    return mergedNode;
}

File: 0617/0617.py
from typing import Optional
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def mergeTrees(self, root1: Optional[TreeNode], root2: Optional[TreeNode]) -> Optional[TreeNode]:
        if (not root1):
            return root2
        if (not root2):
            return root1

        mergedNode = TreeNode()
        mergedNode.val = root1.val + root2.val
        mergedNode.left = self.mergeTrees(root1.left, root2.left)
        mergedNode.right = self.mergeTrees(root1.right, root2.right)

        return mergedNode